docs/source/usage/deployment.rst

```diff
@@ -0,0 +1,140 @@
+  2. Navigate to “Services” and under “Management and Governance” select “Systems Manager”
+  3. Once the page loads, under “Application Management, choose “Parameter Store"
+  4. Create a parameter there in the form of:
+
+      “/<project>/<service>/<environment>/auth_token”
+
+      **Example:** For our Zoom example, this would be something like:
+       
+    .. image:: ../images/Parameter_Store_Example.png
+       :width: 400
+       :alt: /MozDef-EF/zoom/prod/auth_token
+
+  5. In the description field, add “Authentication token used by the Webhook app to post events”.
+  6. For type, select “SecureString”. 
+  7. For value, paste the value from the app’s configuration from your webhook's configured authentication token.
+  8. Add a Tag with key: Project and value: MozDef-EF-<environment>, then create the parameter.
+
+*************************************
+Fill in the Framework Config Template
+*************************************
+
+The following should be done in your local copy of the framework you cloned or forked:
+
+
+  1. Open the downloaded repository in an IDE to edit locally.
+  2. Under config directory, edit “buildspec-prod.yml” file to contain:
+
+    .. code-block:: yaml
+    
+       version: 0.2
+       
+       env:
+         variables:
+           STAGE: prod
+           SERVICE: zoom
+           PROJECT: MozDef-EF
+           API_PATH: zoom
+           TOKEN_ARN: arn:aws:ssm:<REGION_NAME>:<ACCOUNT_ID>:parameter/<parameter-name>
+       
+       phases:
+         install:
+           runtime-versions:
+             python: 3.7
+             nodejs: 10
+           commands:
+             # Install dependencies here
+             - pip3 install --upgrade awscli -q
+             - pip3 install --upgrade pytest -q
+             - pip3 install --upgrade moto -q
+             - pip3 install --upgrade aws-xray-sdk -q
+             - npm install -g --silent --progress=false serverless
+             - npm install --silent --save-dev serverless-pseudo-parameters
+             - npm install --silent --save-dev serverless-prune-plugin
+         pre_build:
+           commands:
+             # Perform pre-build actions here
+             - chmod +x $CODEBUILD_SRC_DIR/config/deploy.sh
+             - $CODEBUILD_SRC_DIR/config/deploy.sh unit-test
+         build:
+           commands:
+             # Invoke the deploy script here
+             - $CODEBUILD_SRC_DIR/config/deploy.sh deploy $STAGE $AWS_REGION
+
+
+  3. The important part here is the filling in of the “env” section at the top of the file. These environment variables will be used by the “serverless.yml” file when deployed by the serverless framework.
+     For each service deployed for a source (such as zoom), the service name and API path will be different.
+  4. Save the file.
+  5. Make any other desired changes on the local copy. For webhook based services, like zoom, there should not be any additional changes needed.
+
+*********************
+Deploy Your Framework
```

Comment:
I know I had written the bulk of this section originally, but this whole section assumes we are using the single source template which uses CodeCommit to house all code + config. And I did so because that's our proposed method to deploy our instance.

However, we should probably document all 3 deployment steps, as there are some important points we don't currently touch here. For instance: if a Github source is used as a single repo, we use a Github personal access token to poll for changes in Github to then trigger a build (and we are using Secrets Manager to store that token in AWS). Naturally none of that is mentioned here.

Another example here, the very first step below actually does not apply if we are using the multi-source template to deploy, as in that case the CodeCommit repo is created by the CloudFormation template during stack creation.

I can help documenting the other deployment steps, but need to actually perform them again to refresh my memory :)

Reply:
I would definitely like some help in mapping out the differences, I think it would be beneficial to demonstrate what those are.

Reply:
I am going to give this a go for other deployment options.

Reply:
In the last 2 commits, I gave this one a go. It's not complete yet, as I have not documented the deployment steps for multi-source scenario (with merge).

_Disclaimer: I suck at editing `.rst` files, I tried to use a preview in VS Code when making the edits. Apologies if it looks crappy, but with these commits I went for content instead of aesthetics._

Reply:
No worries :) we can polish before merging! You're awesome!

Reply:
So I noticed that you listed 'public_repo' as a permission for the github token, if the user has elected to use a private repo, then this would not work, correct?

Reply:
I don't think so. But I think I missed adding a scope, the parent `repo` scope, for which the description in Github says: "Full control of private repositories", as below. I will add this.

![image](https://user-images.githubusercontent.com/30646014/70481729-a09cc000-1b37-11ea-93ad-5a3b2d8b69ab.png)

Reply:
I will add a pic instead to the relevant section in the next commit.